src/Network/GRPC/Common/HTTP2Settings.hs

```diff
@@ -0,0 +1,95 @@
+-- | Settings and parameters pertaining to HTTP\/2
+--
+-- Intended for unqualified import.
+
+module Network.GRPC.Common.HTTP2Settings
+  ( HTTP2Settings(..)
+  , defaultHTTP2Settings
+  ) where
+
+import Data.Default
+import Data.Word
+
+-- | HTTP\/2 settings
+data HTTP2Settings = HTTP2Settings {
+      -- | Maximum number of concurrent active streams
+      --
+      -- <https://datatracker.ietf.org/doc/html/rfc7540#section-5.1.2>
+      http2MaxConcurrentStreams :: Word32
+
+      -- | Initial window size for streams
+      --
+      -- <https://datatracker.ietf.org/doc/html/rfc7540#section-6.9.2>
+    , http2InitialStreamWindowSize :: Word32
+
+      -- | Initial connection window size
+      --
+      -- This setting is specific to the
+      -- [http2](https://hackage.haskell.org/package/http2) package's
+      -- implementation of the HTTP\/2 specification. In particular, this value
+      -- will be broadcast via a @WINDOW_UDPATE@ frame at the beginning of a new
+      -- connection.
+      --
+      -- If the consumed window space of all streams exceeds this value, the
+      -- sender will stop sending data. Therefore, if this value is less than
+      -- @'http2MaxConcurrentStreams' * 'http2InitialStreamWindowSize'@, there
+      -- is risk of a control flow deadlock, since the connection window space
+      -- may be used up by streams that we are not yet processing before we have
+      -- received all data on the streams that we /are/ processing. To reduce
+      -- this risk, increase 'serverNumberOfWorkers'. See
+      -- <https://github.com/kazu-yamamoto/network-control/pull/4> for more
+      -- information.
+    , http2InitialConnectionWindowSize :: Word32
```

Comment:
Same comment here regarding "initial".

I'm not quite sure why you are listing this one as specific to `http2` (perhaps due to an old comment of mine in the `grapesy` codebase, which I wrote prior to understanding all of this? :smile: ). The connection window size is not specific to `http2`-the-package, it's part of the spec; I couldn't find a specific section about the "connection window", but we could refer to https://datatracker.ietf.org/doc/html/rfc7540#section-6.9.1 for example. While it is true that `http2` sets this up at the start of the connection, I think that's something we don't to mention here (I think that text probably came from my old comment).

I really like the second paragraph, very clear.